src/pip/_internal/download.py

```diff
@@ -181,6 +290,19 @@ def __call__(self, req):
 
         return req
 
+    def _prompt_for_password(self, netloc):
+        # Factored out to allow for easy patching in tests
```

Comment:
I would put this comment before the function (since it is about the function as a whole rather than commenting on the implementation inside). And similar for `_should_save_password_to_keyring()` below.